dartagnan/src/main/java/com/dat3m/dartagnan/expression/type/TypeFactory.java

```diff
@@ -92,7 +109,63 @@ public IntegerType getByteType() {
     }
 
     public int getMemorySizeInBytes(Type type) {
-        return TypeLayout.of(type).totalSizeInBytes();
+        return getMemorySizeInBytes(type, true);
+    }
+
+    public int getMemorySizeInBytes(Type type, boolean padded) {
+        if (type instanceof BooleanType) {
+            return 1;
+        }
+        if (type instanceof IntegerType integerType) {
+            return IntMath.divide(integerType.getBitWidth(), 8, RoundingMode.CEILING);
+        }
+        if (type instanceof FloatType floatType) {
+            return IntMath.divide(floatType.getBitWidth(), 8, RoundingMode.CEILING);
+        }
+        if (type instanceof ArrayType arrayType) {
+            if (arrayType.hasKnownNumElements()) {
+                Type elType = arrayType.getElementType();
+                return getMemorySizeInBytes(elType) * arrayType.getNumElements();
+            }
+            return -1;
+        }
+        if (type instanceof AggregateType aType) {
+            List<TypeOffset> typeOffsets = aType.getTypeOffsets();
+            if (typeOffsets.isEmpty()) {
+                return 0;
+            }
+            if (aType.getTypeOffsets().stream().anyMatch(o -> getMemorySizeInBytes(o.type()) < 0)) {
+                return -1;
+            }
```

Comment:
Can this if ever really happen? When would we return -1 for the size in bits of a type?